pkg/volume/plugins.go

```diff
@@ -1027,6 +1027,7 @@ func (pm *VolumePluginMgr) Run(stopCh <-chan struct{}) {
 		// start informer for CSIDriver
 		informerFactory := kletHost.GetInformerFactory()
 		informerFactory.Start(stopCh)
+		informerFactory.WaitForCacheSync(stopCh)
```

Comment:
Hm, I believe that if kubelet runs in standalone mode, then having `WaitForCacheSync` here is problematic because there's no apiserver.

Reply:
this is guarded by the `if` here
https://github.com/kubernetes/kubernetes/blob/941e9242a50fa8f273a0582eca283176c64e3085/pkg/kubelet/volumemanager/volume_manager.go#L271-L274

Reply:
Ah yes, that should protect it, but I wonder if we need to change the ordering of it because right now the reconciler (which is the one that will call into MountVolume) starts before this.

Reply:
done.

Reply:
looks like we should not block on the `WaitForCacheSync` before reconciler because it will timeout the bootstrap:

```
Nov 04 21:48:15 kind-control-plane kubelet[266]: E1104 21:48:15.835859     266 kubelet.go:1650] Unable to attach or mount volumes for pod "kube-apiserver-kind-control-plane_kube-system(b9476abec6e5f541a230b24626dd0325)": unmounted volumes=[k8s-certs usr-local-share-ca-certificates usr-share-ca-certificates ca-certs etc-ca-certificates], unattached volumes=[k8s-certs usr-local-share-ca-certificates usr-share-ca-certificates ca-certs etc-ca-certificates]: timed out waiting for the condition; skipping pod
```

Reply:
Thanks for experimenting around. I opened up https://github.com/kubernetes/kubernetes/issues/96316.

For this feature, if there isn't any negative effect from the informer being not in sync, then I'm ok with not waiting on it, and letting a retry resolve itself.

Reply:
`volumeManager.Run` is called in a [go routine](https://github.com/kubernetes/kubernetes/blob/941e9242a50fa8f273a0582eca283176c64e3085/pkg/kubelet/kubelet.go#L1387) so we won't block kubelet boostrapping.